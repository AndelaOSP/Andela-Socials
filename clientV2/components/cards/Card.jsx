import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/components/_card.scss';

/**
 * @description - This component displays a reusable card
 *
 * @param {object} props { children, style }
 *
 * @returns JSX
 */
const Card = (props) => {
  const {
    children, style,
  } = props;

  return (
    <div
      className="card"
      style={style}
    >
      {children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.objectOf(PropTypes.string),
};

Card.defaultProps = { style: {} };

export default Card;
