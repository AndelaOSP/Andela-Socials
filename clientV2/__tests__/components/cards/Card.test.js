
import React from 'react';

import { shallow } from 'enzyme';

import Card from '../../../components/cards/Card';

describe('Card component', () => {
  it('should render correctly', () => {
    const props = {
      style: {
        height: '10px',
        width: '120px',
      },
      children: '<div>code block</div>',
    };
    const wrapper = shallow(<Card { ...props }/>);

    expect(wrapper).toMatchSnapshot();
  });
});
