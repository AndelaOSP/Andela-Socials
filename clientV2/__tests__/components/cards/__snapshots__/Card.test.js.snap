// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    style={
      Object {
        "height": "10px",
        "width": "120px",
      }
    }
  >
    &lt;div&gt;code block&lt;/div&gt;
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "<div>code block</div>",
      "className": "card",
      "style": Object {
        "height": "10px",
        "width": "120px",
      },
    },
    "ref": null,
    "rendered": "<div>code block</div>",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "<div>code block</div>",
        "className": "card",
        "style": Object {
          "height": "10px",
          "width": "120px",
        },
      },
      "ref": null,
      "rendered": "<div>code block</div>",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
